crates/matrix-sdk-ui/src/timeline/read_receipts.rs

```diff
@@ -116,13 +120,44 @@ impl ReadReceipts {
         //   receipt.
         // - If old_receipt_item is None and new_receipt_item is Some, the new receipt
         //   is likely more recent because it has a place in the timeline.
+
+        // Remove the old receipt from the old event.
+        if let Some(old_event_id) = old_event_id {
+            let is_empty =
+                if let Some(event_receipts) = self.events_read_receipts.get_mut(old_event_id) {
+                    event_receipts.remove(receipt.user_id);
+                    event_receipts.is_empty()
+                } else {
+                    false
+                };
+            // Remove the entry if the map is empty.
+            if is_empty {
+                self.events_read_receipts.remove(old_event_id);
+            }
```

Comment:
Can you inline `if is_empty` into the `if let` branch and remove the variable entirely?

Reply:
I thought this wouldn't work because we hold a reference to a value inside the map. But it seems the compiler is smart enough to drop it first.